src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1015,30 +1153,20 @@ private void ImportStoreIndirect(int token)
         private void ImportStoreIndirect(TypeDesc type)
         {
             StackEntry value = _stack.Pop();
-            StackEntry destinationPointer = _stack.Pop();
-            LLVMTypeRef requestedPointerType = LLVM.PointerType(GetLLVMTypeForTypeDesc(type), 0);
-            LLVMValueRef typedValue = value.LLVMValue;
-            LLVMValueRef typedPointer = destinationPointer.LLVMValue;
+            var destinationPointer = _stack.Pop() as ExpressionEntry;
```

Comment:
If this isn't an ExpressionEntry, we'll fail will a NullReferenceException later. It would be preferable to just do a cast here for the right exception.

Reply:
Also I don't think this is guaranteed to be an ExpressionEntry. For example:
```
ldnull
stind
```
Should be a legal sequence (albeit one that throws a NullReferenceException)